runtime/centrifuge/src/migrations.rs

```diff
@@ -9,5 +9,164 @@
+
+mod asset_registry {
+	use cfg_types::{tokens as v1, tokens::CustomMetadata};
+	#[cfg(feature = "try-runtime")]
+	use frame_support::ensure;
+	use frame_support::{pallet_prelude::OptionQuery, storage_alias, Twox64Concat};
+	use orml_traits::asset_registry::AssetMetadata;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	use super::*;
+	use crate::VERSION;
+
+	/// Migrate all the registered asset's metadata to the new version of
+	/// `CustomMetadata` which contains a `CrossChainTransferability` property.
+	/// At this point in time, the `transferability` of Tranche tokens should be
+	/// set to `CrossChainTransferability::Xcm` and for all other tokens to
+	/// `CrossChainTransferability::Xcm`, with the exception of
+	/// `Currency::Staking` tokens which are not registered in the first place.
+	pub struct CrossChainTransferabilityMigration;
+
+	// The old orml_asset_registry Metadata storage using v0::CustomMetadata
+	#[storage_alias]
+	type Metadata<T: orml_asset_registry::Config> = StorageMap<
+		orml_asset_registry::Pallet<T>,
+		Twox64Concat,
+		CurrencyId,
+		AssetMetadata<Balance, v0::CustomMetadata>,
+		OptionQuery,
+	>;
+
+	impl OnRuntimeUpgrade for CrossChainTransferabilityMigration {
+		fn on_runtime_upgrade() -> Weight {
+			// todo(nuno): this migration will **most likely** be part of the set of
+			// migrations when upgrading to Centrifuge 1020. Refine when we know
+			// for sure.
+			if VERSION.spec_version != 1020 {
+				return Weight::zero();
+			}
+
+			orml_asset_registry::Metadata::<Runtime>::translate(
+				|asset_id: CurrencyId, old_metadata: AssetMetadata<Balance, v0::CustomMetadata>| {
+					match asset_id {
+						CurrencyId::Staking(_) => None,
+						CurrencyId::Tranche(_, _) => Some(to_metadata_v1(
+							old_metadata,
+							v1::CrossChainTransferability::Connectors,
+						)),
+						_ => Some(to_metadata_v1(
+							old_metadata,
+							v1::CrossChainTransferability::Xcm,
+						)),
+					}
+				},
+			);
+
+			let n = orml_asset_registry::Metadata::<Runtime>::iter().count() as u64;
+			<Runtime as frame_system::Config>::DbWeight::get().reads_writes(n, n)
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			use codec::Encode;
+
+			let old_state: Vec<(CurrencyId, AssetMetadata<Balance, v0::CustomMetadata>)> =
+				Metadata::<Runtime>::iter().collect::<Vec<_>>();
+
+			Ok(old_state.encode())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(old_state_encoded: Vec<u8>) -> Result<(), &'static str> {
+			use codec::Decode;
+
+			use crate::OrmlAssetRegistry;
+
+			let old_state = sp_std::vec::Vec::<(
+				CurrencyId,
+				AssetMetadata<Balance, v0::CustomMetadata>,
+			)>::decode(&mut old_state_encoded.as_ref())
+			.map_err(|_| "Error decoding pre-upgrade state")?;
+
+			for (asset_id, old_metadata) in old_state {
+				let new_metadata = OrmlAssetRegistry::metadata(asset_id)
+					.ok_or_else(|| "New state lost the metadata of an asset")?;
```

Comment:
Eagle eyes! If won't return an error because we don't have (and assume that maybe too implicitly?) any `CurrencyId::Staking` registered in the registry, so that edge case should never occur.

Reply:
Should be fine then!